osquery/tables/system/darwin/sysctl_utils.cpp

```diff
@@ -76,8 +77,35 @@ void genControlInfo(int* oid,
   size_t oid_type = 0;
   if (response_size > 0) {
     oid_type = ((size_t)response[0] & CTLTYPE);
+    if ((oid_type == 0 || oid_type == CTLTYPE_INT) && response_size > 4) {
+      // For whatever reason, macOS defines fewer CTLTYPE's than BSD, and
+      // sometimes uses the format character instead of (or in addition to)
+      // the CTLTYPE to specify the type. Here we detect a few such cases and
+      // map them to CTLTYPE's.
+      // TODO: Both CTLTYPE_INT and CTLTYPE_QUAD can be specified as unsigned
+      // using a similar method.
+      char type_char = response[4];
+      switch (type_char) {
```

Comment:
Should this have a default case?

Reply:
The default is just to leave it as what it was. I can add a comment to that effect.